### PR TITLE
reporting_method = mqtt-smarthome

### PR DESCRIPTION
**Pull request by hobbyquaker**
implements #6

### CODE REVIEW

File: miflora-mqtt-daemon.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import json
 import os.path
 import argparse
+import time
 from time import sleep, localtime, strftime
```
Comment (ThomDietrich) [inline miflora-mqtt-daemon.py]: how about `from time import time, sleep, localtime, strftime` ?

File: miflora-mqtt-daemon.py [hunk 2]
```diff
@@ -156,6 +160,7 @@ def flores_to_openhab_items(flores, reporting_mode):
         print_line('MQTT connection error. Please check your settings in the configuration file "config.ini"', error=True, sd_notify=True)
         sys.exit(1)
     else:
+        mqtt_client.publish('{}/connected'.format(base_topic), payload='1', retain=True)
```
Comment (ThomDietrich) [inline miflora-mqtt-daemon.py]: This one line is unconditional...

File: miflora-mqtt-daemon.py [hunk 3]
```diff
@@ -279,16 +284,22 @@ def flores_to_openhab_items(flores, reporting_mode):
...
                 flora['name_pretty'], flora['mac'], flora['stats']['success']/flora['stats']['count']
                 ), error = True, sd_notify = True)
             print()
+            if reporting_mode == 'mqtt-smarthome':
+                mqtt_client.publish('{}/connected'.format(base_topic), payload='1', retain=True)
             continue
         else:
             flora['stats']['success'] = flora['stats']['success'] + 1
+            if reporting_mode == 'mqtt-smarthome':
+                mqtt_client.publish('{}/connected'.format(base_topic), payload='2', retain=True)
```
Comment (ThomDietrich) [inline miflora-mqtt-daemon.py]: Are these three additions necessary? It's quite common for one sensor to be available only every other try. I'm even seeing a 20% success rate on one of my (deliberately) bad placed sensors.

Are these updates from 1->2->1->2 really meaningful? I'd prefer if we can leave out mode specific routines in most of the daemons parts (Which I'd like to separate into functions at some point)
In my understanding the connected state could be 0 for offline, 1 for initializing and 2 after the Initialization was successful. Your proposal defines 2 as "connected to MQTT and hardware" and I think it's fitting this way. Wdyt?

File: miflora-mqtt-daemon.py (hunk 1 above)
Comment (hobbyquaker) [reply miflora-mqtt-daemon.py]: oh, didn't realize that there are already imports from time... will change that.

File: miflora-mqtt-daemon.py (hunk 3 above)
Comment (hobbyquaker) [reply miflora-mqtt-daemon.py]: hmm yes you're right. There is also another point: the connected topic normally indicates the successful connection to some kind of gateway. In this project we don't have the "one" gateway but multiple devices, so it doesn't really make sense. Will change that and just publish `0` and `2`.

**Comment by ThomDietrich** [comment]
On a side note, I'm not sure if it is meaningful to publish momentary states as retained messages but Hey, that's up to you and your mqtt-smarthome.

Also forgive me for asking but where do you see the advantages of mqtt-smarthome vs. Homie? I've helped extend the Homie convention and your opinion about the advantages and disadvantages would be very interesting. Best!

**Comment by hobbyquaker** [comment]
Publishing retained does make sense from my point of view. As an example: Imagine a Webfrontend that connects to the MQTT Broker and wants to show sensor data. Without retain it would have to wait until the next message arrives or it would need some kind of backend that caches states - with retained messages there is no need for that. On MQTT-Smarthome we publish everything retained except for set commands and "one shot events" (like e.g. a button presses).

About the pros and cons of Homie and MQTT-Smarthome: If you want we can talk via phone, I think that's easier. Will send you my number. Main thing why I stick to mqtt-smarthome: It's one year older than homie and I already got it running and we made a bunch of interfaces before the Homie project started. Another big difference is that mqtt-smarthome is kind a "unopinionated"  and doesn't care about meta-data (like descriptions of datapoint types, assignments to rooms or functions and so on). Stuff like this is handled on another layer in the mqtt-smarthome concept, in my case mainly through my script-engine (https://github.com/hobbyquaker/mqtt-scripts)

**Comment by ThomDietrich** [comment]
Hey Sebastian,
just so we are clear, you've removed the many connected state changes and said you are going to stick with 0 and 2 only, the code contains 0 and 1 messages. Which is the right one?

**Comment by hobbyquaker** [comment]
thought about that again and came to the conclusion that `1` fits better :)

**Comment by ThomDietrich** [comment]
Okay then, here we go!

Thanks for the addition! Happy to get further feedback from your side ;)

**Pull request merged**